From 925f26cc2d5964bb5712279b1647c49eab385f11 Mon Sep 17 00:00:00 2001
From: Jason Lowe <jlowe@apache.org>
Date: Tue, 15 Sep 2015 20:21:33 +0000
Subject: [PATCH 1303/2748] YARN-4158. Remove duplicate close for LogWriter in
 AppLogAggregatorImpl#uploadLogsForContainers.
 Contributed by Zhihai Xu

(cherry picked from commit 8c1cdb17a03fec91bfab172fe92b67bdeb9ebfba)

Change-Id: Ifab4c9b87955f56472e6e03f2834c2fe4cdca688
---
 .../logaggregation/AppLogAggregatorImpl.java       |    1 +
 1 file changed, 1 insertion(+)

diff --git a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/containermanager/logaggregation/AppLogAggregatorImpl.java b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/containermanager/logaggregation/AppLogAggregatorImpl.java
index ce2ec7b..5ac0ce3 100644
--- a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/containermanager/logaggregation/AppLogAggregatorImpl.java
+++ b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/containermanager/logaggregation/AppLogAggregatorImpl.java
@@ -291,6 +291,7 @@ private void uploadLogsForContainers(boolean appFinished) {
 
       if (writer != null) {
         writer.close();
+        writer = null;
       }
 
       long currentTime = System.currentTimeMillis();
-- 
1.7.9.5

